instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/JarServiceNameProviderTest.java

```diff
@@ -0,0 +1,116 @@
+        new JarServiceNameProvider(
+            () -> new String[0], prop -> null, JarServiceNameProviderTest::failPath);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes()).isEmpty();
+  }
+
+  @Test
+  void createResource_noJarFileInArgs() {
+    String[] args = new String[] {"-Dtest=42", "-Xmx666m", "-jar"};
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> args, prop -> null, JarServiceNameProviderTest::failPath);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes()).isEmpty();
+  }
+
+  @Test
+  void createResource_processHandleJar() {
+    String[] args =
+        new String[] {"-Dtest=42", "-Xmx666m", "-jar", "/path/to/app/my-service.jar", "abc", "def"};
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> args, prop -> null, JarServiceNameProviderTest::failPath);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes())
+        .hasSize(1)
+        .containsEntry(ResourceAttributes.SERVICE_NAME, "my-service");
+  }
+
+  @Test
+  void createResource_processHandleJarWithoutExtension() {
+    String[] args = new String[] {"-Dtest=42", "-Xmx666m", "-jar", "/path/to/app/my-service"};
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> args, prop -> null, JarServiceNameProviderTest::failPath);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes())
+        .hasSize(1)
+        .containsEntry(ResourceAttributes.SERVICE_NAME, "my-service");
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(SunCommandLineProvider.class)
+  void createResource_sunCommandLine(String commandLine, String jarPath) {
+    Function<String, String> getProperty =
+        key -> "sun.java.command".equals(key) ? commandLine : null;
+    Predicate<Path> fileExists = file -> jarPath.equals(file.toString());
+
+    JarServiceNameProvider serviceNameProvider =
+        new JarServiceNameProvider(() -> new String[0], getProperty, fileExists);
+
+    Resource resource = serviceNameProvider.createResource(config);
+
+    assertThat(resource.getAttributes())
+        .hasSize(1)
+        .containsEntry(ResourceAttributes.SERVICE_NAME, "my-service");
+  }
+
+  static final class SunCommandLineProvider implements ArgumentsProvider {
+
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+      return Stream.of(
+          arguments("/path/to/my-service.jar", "/path/to/my-service.jar"),
+          arguments(
+              "/path to app/with spaces/my-service.jar 1 2 3",
+              "/path to app/with spaces/my-service.jar"),
+          arguments(
+              "/path to app/with spaces/my-service 1 2 3", "/path to app/with spaces/my-service"));
```

Comment:
I was trying this out, and these tests fail on Windows, because later on the `Path` gets normalized with `\`, it's ok to merge as-is though (I need to try again to get the github actions running on windows)

Reply:
Oh, I completely didn't think of this. Great point, I've changed this test to use `Path` instead, hope that helps 🤞